jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ResourceSelector.kt

```diff
@@ -77,11 +74,16 @@ class ResourceSelector<T> private constructor(
             super.setSelectedItem(message("loading_resource.loading"))
 
             val resource = resourceType.invoke()
-            if (resource == null) {
+            val connectionSettings = connectionSettingsSupplier()
```

Comment:
If we don't have this maybe we should throw? like it will never succeed but it also never reports a failure to the user/implementer